plugins/guisupport/guisupport.h

```diff
@@ -53,14 +53,13 @@ class GuiSupport : public QObject
     void registerVariantHandler();
 #if QT_VERSION >= QT_VERSION_CHECK(5, 0, 0)
     void discoverObjects();
-    void updateWindowTitle(QWindow *w);
-    QIcon createIcon(const QIcon &oldIcon, QWindow *w=nullptr);
-    void updateWindowIcon(QWindow *w);
-    void restoreWindowIcon(QWindow *w);
+    void updateWindowTitle(QObject *o);
+    QIcon createIcon(const QIcon &oldIcon, QObject *o=nullptr);
```

Comment:
Hm either remove the default value or still use a QWindow and drop the cast etc as you use qApp directly anyhow .